src/Stratis.Bitcoin.Features.SmartContracts/Rules/AllowedScriptTypeRule.cs

```diff
@@ -92,7 +92,7 @@ private static void CheckInput(Network network, TxIn input)
                 return;
 
             // For cross-chain transfers
-            if (PayToMultiSigTemplate.Instance.CheckScriptSig(network, input.ScriptSig, null))
+            if (PayToFederationTemplate.Instance.CheckScriptSig(network, input.ScriptSig, null))
```

Comment:
I think we would need to still allow pay-to-multisig so that the historical Cirrus blocks remain valid?

In general, keeping the widest possible range of allowed script types seems to be desirable.

Reply:
Done.